hrm-service/src/csam-report/csam-report-routes-v0.ts

```diff
@@ -9,10 +10,35 @@ const csamReportRouter = SafeRouter();
 csamReportRouter.post('/', publicEndpoint, async (req: Request & { accountSid: string }, res: Response) => {
   const { accountSid } = req;
 
-  const { contactId, csamReportId, twilioWorkerId } = req.body;
+  const { contactId, csamReportId, twilioWorkerId, reportType } = req.body;
 
-  const createdCSAMReport = await createCSAMReport({ contactId, csamReportId, twilioWorkerId }, accountSid);
+  // Validate that the payload has a proper format
+  if (reportType && reportType !== "counsellor-generated" && reportType !== "self-generated") {
+    throw createError(422, 'Invalid argument "reportType" provided'); 
+  }
+  if ((!reportType || reportType === "counsellor-generated") && !csamReportId) {
+    throw createError(422, 'Invalid, "reportType" argument specifies "counsellor-generated" report, but no csamReportId argument provided'); 
+  }
+
+  const createdCSAMReport = await createCSAMReport({ contactId, csamReportId, twilioWorkerId, reportType }, accountSid);
   res.json(createdCSAMReport);
 });
 
+csamReportRouter.post('/:reportId/acknowledge', publicEndpoint, async (req: Request & { accountSid: string }, res: Response) => {
```

Comment:
```suggestion
csamReportRouter.post('/:reportId(\\d+)/acknowledge', publicEndpoint, async (req: Request & { accountSid: string }, res: Response) => {
```

We wouldn't have to validate the ID is a number in code then?